src/components/EggheadPlayer/use-egghead-player.ts

```diff
@@ -70,9 +70,12 @@ const storeProgress = (
   })
 }
 
-const onProgress = (lesson: LessonResource) => async (progress: {
-  playedSeconds: any
-}) => {
+const onProgress = async (
```

Comment:
instead of trying to do a "double jump" we require the lesson to be passed in explicitly